### ISSUE TITLE
Typescript Bindings: Go function that returns error creates Promise<Error> instead of Promise<void>

### ISSUE THREAD

**Issue by ATenderholt** [wailsapp/wails#2242]
### Description

If there is a bound go function that returns an error, it creates a TypeScript binding that returns a `Promise<Error>` instead of `Promise<void>`.

### To Reproduce

1. Create a go function that returns an error (e.g. `func SomeFunc() error`).
2. Run `wails dev` or `wails generate module`.
3. Generated Typescript will be `export function SomeFunc():Promise<Error>;`.

### Expected behaviour

Expecting `Promise<void>` so that the promise behavior is consistent with rest of JavaScript, and the corresponding error would be handled in the `.catch()` instead of `.then()`.

### System Details

```shell
Wails CLI v2.2.0

Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:		MacOS
Version: 	12.6.1
ID:		21G217
Go Version:	go1.19.2
Platform:	darwin
Architecture:	arm64

Wails
------
Version: 	v2.2.0

Dependency			Package Name	Status		Version
----------			------------	------		-------
xcode command line tools  	N/A 		Installed 	2395
npm  				N/A 		Installed 	8.19.2
*upx  				N/A 		Available 	
*nsis  				N/A 		Available 	

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details: 
  - upx : Available at https://upx.github.io/
  - nsis : Available at https://nsis.sourceforge.io/Download



If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

**Comment by leaanthony** [comment]
That's a valid point! Nice catch. Is this something you're happy to address yourself?

**Comment by ATenderholt** [comment]
Yep, I think I can take care of it within the next few days. I'll let you know if I get stuck.

**Comment by ATenderholt** [comment]
Addressed w/ PR #2247.
